Dockerfile

```diff
@@ -24,11 +24,12 @@ USER root
 ENV HOME=/root
 
 # Install build and dev tools
-# NOTE: Require python38 to install pre-commit
+# NOTE: Require python311 to install pre-commit
 RUN --mount=type=cache,target=/root/.cache/dnf:rw \
     dnf install --setopt=cachedir=/root/.cache/dnf -y --nodocs \
         nodejs \
-        python38 \
+        python3.11 \
+        python3.11-pip \
```

Comment:
do we need to install pip ?

Reply:
3.8 was adding it, it seems that on newer Python versions it was moved to its package.
It is needed by the per-commit.

Note, that it keeps the image exactly as it was before.

Reply:
ok